src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Operators.cs

```diff
@@ -2684,6 +2684,9 @@ private static AggregateType GetEnumBinOpType(ExpressionKind ek, CType argType1,
             AggregateType type1 = argType1 as AggregateType;
             AggregateType type2 = argType2 as AggregateType;
 
+            if (type1 == null)
+                throw new Exception("invalid cast argType1 as AggregateType");
+
```

Comment:
Same here, when will this be hit?

Reply:
We can't analyze the entire chain of calls, but this code refers to a variable resulting from a conversion without first checking whether the conversion was successful